tests/test_awaitable_task.py

```diff
@@ -0,0 +1,89 @@
+from aiida_workgraph import WorkGraph, task
+import asyncio
+from aiida.cmdline.utils.common import get_workchain_report
+import datetime
+import os
+import pytest
+
+
+def test_awaitable_task(decorated_add):
+    @task.awaitable()
+    async def awaitable_func(x, y):
+        n = 2
+        while n > 0:
+            n -= 1
+            await asyncio.sleep(0.5)
+        return x + y
+
+    wg = WorkGraph(name="test_awaitable")
+    awaitable1 = wg.add_task(awaitable_func, "awaitable_func1", x=1, y=2)
+    add1 = wg.add_task(decorated_add, "add1", x=1, y=awaitable1.outputs["result"])
+    wg.run()
+    report = get_workchain_report(wg.process, "REPORT")
+    assert "Waiting for child processes: awaitable_func1" in report
+    assert add1.outputs["result"].value == 4
+
+
+def test_time_monitor(decorated_add):
+    """Test the time monitor task."""
+    wg = WorkGraph(name="test_time_monitor")
+    monitor1 = wg.add_task(
+        "workgraph.time_monitor",
+        "monitor1",
+        datetime=datetime.datetime.now() + datetime.timedelta(seconds=10),
+    )
+    add1 = wg.add_task(decorated_add, "add1", x=1, y=2)
+    add1.waiting_on.add(monitor1)
+    wg.run()
+    report = get_workchain_report(wg.process, "REPORT")
+    assert "Waiting for child processes: monitor1" in report
+    assert add1.outputs["result"].value == 3
+
+
+def test_file_monitor(decorated_add):
+    """Test the file monitor task."""
+
+    @task.awaitable()
+    async def create_test_file(filepath="/tmp/test_file_monitor.txt", t=2):
+        await asyncio.sleep(t)
+        with open(filepath, "w") as f:
+            f.write("test")
+
+    # remove the test file if it exists
+    try:
+        os.remove("/tmp/test_file_monitor.txt")
+    except FileNotFoundError:
+        pass
+
+    wg = WorkGraph(name="test_file_monitor")
+    monitor1 = wg.add_task(
+        "workgraph.file_monitor", name="monitor1", filepath="/tmp/test_file_monitor.txt"
+    )
+    wg.add_task(create_test_file, "create_test_file1")
+    add1 = wg.add_task(decorated_add, "add1", x=1, y=2)
+    add1.waiting_on.add(monitor1)
+    wg.run()
+    report = get_workchain_report(wg.process, "REPORT")
+    assert "Waiting for child processes: monitor1" in report
+    assert add1.outputs["result"].value == 3
```

Comment:
pytest has the `tmp_path` fixture that is always autouse that cleans up the file https://docs.pytest.org/en/stable/how-to/tmp_path.html after the run of the test.
```suggestion
def test_file_monitor(decorated_add, tmp_path):
    """Test the file monitor task."""

    monitor_file_path = tmp_path / "test_file_monitor.txt"
    @task.awaitable()
    async def create_test_file(filepath=monitor_file_path, t=2):
        await asyncio.sleep(t)
        with open(filepath, "w") as f:
            f.write("test")

    wg = WorkGraph(name="test_file_monitor")
    monitor_task = wg.add_task(
        "workgraph.file_monitor", name="monitor_task", filepath=monitor_file_path
    )
    wg.add_task(create_test_file, "create_test_file_task")
    add_task = wg.add_task(decorated_add, "add_task", x=1, y=2)
    add_task.waiting_on.add(monitor_task)
    wg.run()
    report = get_workchain_report(wg.process, "REPORT")
    assert "Waiting for child processes: monitor_task" in report
    assert add_task.outputs["result"].value == 3
```

Reply:
I didn't know this. Thanks!